### PR TITLE
Initial content rewrite

### PR DESCRIPTION
**Pull request by kenrogers**
## Initial content rewrite for docs

- Rewrote a significant portion of the docs to add information and make things more clear
- Added multiple sections like Stacks Academy, Tutorials, and Cookbook

### CODE REVIEW

File: docs/gaia/gaia.md
Comment (wileyj) [inline docs/gaia/gaia.md:15]: `censorship-resistant, permanent storage` -> `censorship-resistant permanent storage`

File: docs/intro.mdx
Comment (wileyj) [inline docs/intro.mdx:1]: why is this file an `mdx` file, vs others that are `md`? is there something special about this file where it needs mdx?

File: docs/next-steps/index.md
Comment (wileyj) [inline docs/next-steps/index.md:12]: i would remove `View` here, but that's an opinionated change. I don't feel it adds to the description

File: docs/nodes-and-miners/miner-mainnet.md
Comment (wileyj) [inline docs/nodes-and-miners/miner-mainnet.md:1]: I'm a little biased here, but the docs i created in stacksfoundation/miner-docs should be treated as the "source of truth" for this type of content. the content from the "prod" docs is a bit outdated. 

I'm not a fan of copy/pasting the content from there here, but i don't see a good alternative just yet

File: docs/nodes-and-miners/miner-testnet.md
Comment (wileyj) [inline docs/nodes-and-miners/miner-testnet.md:1]: see the comment above - it's the same docs, just for testnet rather than mainnet

File: docs/nodes-and-miners/run-a-node.md
Comment (wileyj) [inline docs/nodes-and-miners/run-a-node.md:1]: same as the miner content - a lot of changes have been made to the docs in the repo itself, will hae to verify if this is still correct. 

i want to find a way for the docs to essentially "import" external docs we want to use

Comment (wileyj) [reply docs/nodes-and-miners/run-a-node.md:1]: Can confirm that the content shown here is incorrect. will have to sendyou an update or merge this to the docs repo and update before it's merged to master

File: docs/stacks-academy/sips.md
Comment (wileyj) [inline docs/stacks-academy/sips.md:1]: I would add a third `tip` here, linking to the sips call event on the calendar

**Comment by wileyj** [comment]
I would also like to see a dedicated section/page with more prominence given to the whitepapers

**Comment by wileyj** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/nodes-and-miners/miner-costs-and-fees


I think we should add a link to friedgers cost tool here - https://friedger.id/mining-calculator/

**Comment by wileyj** [comment]
Screenshot 2023-01-09 at 15 47 29


Not a huge fan of how cluttered this top bar looks. do we need all of these at the top, or could they be given higher prominence on the landing page perhaps?

**Comment by satoshibles** [comment]
Will we be adding this section back?

https://docs.stacks.co/docs/services-using-stacks/nft

File: docs/tutorials/hello-stacks.md
Comment (Hero-Gamer) [inline docs/tutorials/hello-stacks.md:212]: spelling error: pringing
correct word: bringing

File: docs/tutorials/hello-stacks.md
Comment (Hero-Gamer) [inline docs/tutorials/hello-stacks.md:224]: error: prifix
correct word: prefix

File: docs/tutorials/hello-stacks.md
Comment (Hero-Gamer) [inline docs/tutorials/hello-stacks.md:272]: correct word: mnemonic

File: docs/tutorials/hello-stacks.md
Comment (Hero-Gamer) [inline docs/tutorials/hello-stacks.md:725]: Incorrect: obvisouly
Correct: obviously
(sorry I don't know how to use Github properly so I'm not sure if I'm submitting these proposed changes correctly, please do tell me if there a better way you wish me to submit)

File: docs/tutorials/hello-stacks.md
Comment (Hero-Gamer) [inline docs/tutorials/hello-stacks.md:735]: incorrect: developmenet 
correct: development

File: docs/stacks-academy/what-is-stacks.md
Comment (Hero-Gamer) [inline docs/stacks-academy/what-is-stacks.md:67]: incorrect: ultiamtely 
correct: ultimately

**Comment by Hero-Gamer** [comment]
I can't find the content under "files changed" tab for this page: https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/stacks-academy/sips

Error on this page is on line SIP 004, the word "Committment" is spelled wrong,
should be "Commitment"

File: docs/stacks-academy/proof-of-transfer.md
Comment (Hero-Gamer) [inline docs/stacks-academy/proof-of-transfer.md:63]: incorrect: bllock (at the end of the paragraph)
correct: block

File: docs/stacks-academy/proof-of-transfer.md
Comment (Hero-Gamer) [inline docs/stacks-academy/proof-of-transfer.md:65]: incorrect: occured
correct: occurred

**Comment by Hero-Gamer** [comment]
again, can't find the text from this page https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/stacks-academy/microblocks  under the Files Changed tab. 

under **Enabling microblocks**
Miners can choose to enable or disable microblocks in their mining configuration. As a best practice, miners should enable microblock mining. 

that last microblock word should be plural I think, therefore should say **microblocks**

whole paragraph should say:
**_Enabling microblocks
Miners can choose to enable or disable microblocks in their mining configuration. As a best practice, miners should enable microblocks mining._**

File: docs/stacks-academy/post-conditions.md
Comment (Hero-Gamer) [inline docs/stacks-academy/post-conditions.md:8]: incorrect: adn
correct: and

Comment (Hero-Gamer) [inline docs/stacks-academy/post-conditions.md:8]: incorrect: arechitecture 
correct: architecture

File: docs/stacks-academy/stacking.md
Comment (Hero-Gamer) [inline docs/stacks-academy/stacking.md:123]: incorrect: stategy.
correct: strategy.

File: docs/stacks-academy/testnet.md
Comment (Hero-Gamer) [inline docs/stacks-academy/testnet.md:7]: image
incorrect: environnment
correct: environment

File: docs/stacks-academy/technical-specs.md
Comment (Hero-Gamer) [inline docs/stacks-academy/technical-specs.md:79]: incorrect: whitepater
correct whitepaper

File: docs/clarity/index.md
Comment (Hero-Gamer) [inline docs/clarity/index.md:10]: incorrect: langauge
correct: language

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1744]: Incorrect: successfuly
Correct: successfully

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1768]: Incorrect: successfuly
Correct: successfully

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1814]: Incorrect: successfuly
Correct: successfully

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1844]: Incorrect: successfuly
Correct: successfully

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1765]: Incorrect: transfered 
Correct: transferred

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1811]: Incorrect: transfered 
Correct: transferred

File: docs/clarity/language-functions.md
Comment (Hero-Gamer) [inline docs/clarity/language-functions.md:1427]: Sure it's "destructure" right? not "restructure"?
Just flagging in case.

File: docs/clarity/noteworthy-contracts/bns-contract.md
Comment (Hero-Gamer) [inline docs/clarity/noteworthy-contracts/bns-contract.md:208]: Incorrect: curretly
Correct: currently

File: docs/clarity/noteworthy-contracts/xbtc.md
Comment (Hero-Gamer) [inline docs/clarity/noteworthy-contracts/xbtc.md:22]: Incorrect: currrently
Correct: currently

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/nodes-and-miners/run-a-node
Under the sub-page: **run-a-node with DOCKER**

Incorrect: prefered
Correct: preferred

Possible incorrect: Vcpu
Possible correct: cpu      

Incorrect: perfomance image
Correct: performance

Incorrect: perfomance image
Correct: performance

Incorrect: optionaly 
Correct: optionally

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/nodes-and-miners/digitalocean
Incorrect: prefered
Correct: preferred

File: docs/nodes-and-miners/miner-testnet.md
Comment (Hero-Gamer) [inline docs/nodes-and-miners/miner-testnet.md:129]: incorrect: minig guide
correct: mining guide

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/nodes-and-miners/miner-mainnet
Incorrect: crytpocurrency 
Correct: cryptocurrency

There are 6 same error / occurrences on this page.

**Comment by kenrogers** [comment]
> ping @kenrogers if you need help, maybe add myself and a few others here as collaborators on your branch so we can get these corrections resolved?

Going through and addressing all these today. I also added a spell check extension in VS Code and will be running that through all files to hopefully eliminate that issue. Will add you and Hero as collaborators after my initial sweep today so we don't end up with conflicts on this initial batch of fixes.

File: docs/nodes-and-miners/miner-mainnet.md
Comment (kenrogers) [reply docs/nodes-and-miners/miner-mainnet.md:1]: Leaving this for now per our conversation, with the plan to copy/paste after initial fixes and then find a better solution.

File: docs/nodes-and-miners/miner-testnet.md
Comment (kenrogers) [reply docs/nodes-and-miners/miner-testnet.md:1]: Leaving this for now per our conversation, with the plan to copy/paste after initial fixes and then find a better solution.

File: docs/nodes-and-miners/run-a-node.md
Comment (kenrogers) [reply docs/nodes-and-miners/run-a-node.md:1]: Similar to the above, leaving this for now per our conversation, with the plan to copy/paste after initial fixes and then find a better solution.

**Comment by kenrogers** [comment]
> Will we be adding this section back?
> 
> https://docs.stacks.co/docs/services-using-stacks/nft

Yep, added this back in most recent push.

**Comment by kenrogers** [comment]
Everything in this PR has been addressed except for the Gaia/miner content, which will be addressed with future changes in the `develop` branch.

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/gaia/
Incorrect: immutabile
Correct: immutable

File: docs/gaia/gaia-on-ec2.md
Comment (Hero-Gamer) [inline docs/gaia/gaia-on-ec2.md:29]: Incorrect: appropiate
Correct: appropriate

File: docs/gaia/setup-linux.md
Comment (Hero-Gamer) [inline docs/gaia/setup-linux.md:52]: Incorrect: localy
Correct: locally

File: docs/cookbook/creating-an-ft.md
Comment (Hero-Gamer) [inline docs/cookbook/creating-an-ft.md:14]: https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/cookbook/creating-an-ft
On the website, spelling error is there, but in the "Files Changed" codes, it is not there
But I'm gonna by the website anyways, needs addressing this error. 

Incorrect: Since FTs may be divisible, the [FT official mainnet deployment trait adress]
Correct: Since FTs may be divisible, the [FT official mainnet deployment trait address]

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/contribute/#articles
More of a visual/functional comment: The links that only appears when you hover cursor over felt little strange and pointless? I'm not sure the purpose of it without further context, just flagging.
image

**Comment by Hero-Gamer** [comment]
https://stacks-docs-lcj697nen-stacks-team.vercel.app/docs/contribute/docs
(Some of which below may already been commented already, some may not have)

Incorrect:docs reposiroty
Correct: docs repository

Incorrect: so you can see a preview of yourhanges in real time.
Correct: so you can see a preview of your changes in real time.

Incorrect: Aditional information
Correct: Additional information

Incorrect: alterative
Correct: alternative

File: docs/glossary/index.md
Comment (Hero-Gamer) [inline docs/glossary/index.md:64]: Incorrect: intented
Correct: intended

**Comment by wileyj** [comment]
@kenrogers once you resolve the conflicts, let's merge this. 
@Hero-Gamer please hold off on any new comments until we can get the changes into this repo where we can all make changes as needed vs putting it all on Ken

**Review by wileyj — APPROVED** [review]
still some work to do, but merging this into develop is fine so we can spread the load a little bit.

**Comment by wileyj** [comment]
merge into https://github.com/stacks-network/docs/tree/develop. 

I'll open a PR for develop-> master so we can continue the updates there

**Comment by Hero-Gamer** [comment]
I finished commenting all good! I checked all pages I think :P
Good luck let me know if you need a second run through or other help.

**Pull request merged**
